doc/spelling.md

```diff
@@ -44,7 +44,7 @@ Value specifications take the form:
 
 See [spelling databases](#spelling-databases) for how to build the database part of the name.
 
-The `value-name` part can be either a hash or a dataset name. If  `value-name` matches the pattern `^#sha1-[0-9a-fA-F]{40}$`, it will be interpreted as a hash. Otherwise it will be interpreted as a dataset name.
+The `value-name` part can be either a hash or a dataset name. If  `value-name` matches the pattern `^#[0-9a-fA-F]{40}$`, it will be interpreted as a hash. Otherwise it will be interpreted as a dataset name.
```

Comment:
Also the pattern isn't `0-9a-fA-F`. Maybe just either find the code that does this, or find another doc which specs it out.